src/optimizer.cpp

```diff
@@ -250,24 +274,24 @@ void Optimizer::integrateStateVelocities(
   auto y = xt::view(trajectories, xt::all(), xt::all(), 1);
   auto yaw = xt::view(trajectories, xt::all(), xt::all(), 2);
 
-  auto w = state.getVelocitiesWZ();
+  auto &w = state.wz;
   double initial_yaw = tf2::getYaw(state_.pose.pose.orientation);
   yaw = xt::cumsum(w * settings_.model_dt, 1) + initial_yaw;
-  xt::xtensor<double, 2> yaw_offseted = yaw;
+  xt::xtensor<float, 2> yaw_offseted = yaw;
 
   xt::view(yaw_offseted, xt::all(), xt::range(1, _)) =
     xt::view(yaw, xt::all(), xt::range(_, -1));
   xt::view(yaw_offseted, xt::all(), 0) = initial_yaw;
 
-  auto yaw_cos = xt::eval(xt::cos(yaw_offseted));
-  auto yaw_sin = xt::eval(xt::sin(yaw_offseted));
+  xt::xtensor<float, 2> yaw_cos = xt::cos(yaw_offseted);
```

Comment:
Why remove the evals? I found in some cases that helped actually with the jitter, I suspect because we asked it to do something up front rather than trying to evaluate later with some concatenated operations

Reply:
Mostly, I see improvements w.r.t. evals on the trig functions (cos, sin, atan2)

Reply:
if you assign to tensor it is equal to calling eval on xexpression both make evaluation in place (not lazily)